services/audit-service/src/main/java/datawave/microservice/audit/health/rabbit/RabbitHealthChecker.java

```diff
@@ -0,0 +1,705 @@
+     * @return An {@link Exchange}
+     */
+    private static Exchange createExchange(ExchangeProperties ep) {
+        ExchangeBuilder builder = new ExchangeBuilder(ep.getName(), ep.getType()).durable(ep.isDurable());
+        if (ep.isAutoDelete())
+            builder.autoDelete();
+        if (ep.isDelayed())
+            builder.delayed();
+        if (ep.isInternal())
+            builder.internal();
+        return builder.build();
+    }
+    
+    /**
+     * Creates an {@link Queue} from the configured {@link QueueProperties}
+     * 
+     * @param qp
+     *            The queue properties derived from the {@link RabbitHealthProperties}
+     * @return A {@link Queue}
+     */
+    private static Queue createQueue(QueueProperties qp) {
+        return new Queue(qp.getName(), qp.isDurable(), qp.isExclusive(), qp.isAutoDelete(), qp.getArguments());
+    }
+    
+    /**
+     * Creates an {@link Binding} from the configured {@link BindingProperties}
+     * 
+     * @param bp
+     *            The binding properties derived from the {@link RabbitHealthProperties}
+     * @return A {@link Binding}
+     */
+    private static Binding createBinding(BindingProperties bp) {
+        Binding.DestinationType destType;
+        switch (bp.getDestinationType().toLowerCase()) {
+            case "queue":
+                destType = Binding.DestinationType.QUEUE;
+                break;
+            case "exchange":
+                destType = Binding.DestinationType.EXCHANGE;
+                break;
+            default:
+                throw new RuntimeException("Unable to create binding given properties: [" + bp + "]");
+        }
+        return new Binding(bp.getDestination(), destType, bp.getExchange(), bp.getRoutingKey(), bp.getArguments());
+    }
+    
+    /**
+     * Determines which poll interval should be used, based on the overall RabbitMQ health
+     * 
+     * @return the poll interval to use in millis
+     */
+    @Override
+    public long pollIntervalMillis() {
+        if (isHealthy())
+            return rabbitHealthProperties.getHealthyPollIntervalMillis();
+        else
+            return rabbitHealthProperties.getUnhealthyPollIntervalMillis();
+    }
+    
+    /**
+     * Performs a health check of RabbitMQ.
+     * <p>
+     * This method is synchronized with the isHealthy method to ensure that the health check is given time to complete before deciding whether to accept an
+     * audit request via the {@link AuditController}
+     * <p>
+     * RabbitMQ is considered to be healthy if the cluster, exchanges, queues, and bindings are configured as specified in the {@link RabbitHealthProperties}.
+     * If we are not able to determine the health of the cluster, exchanges, queues, and bindings (perhaps due to the management API being unavailable, or the
+     * cluster being down), we will consider RabbitMQ to be unhealthy.
+     */
+    @Override
+    public void runHealthCheck() {
+        initRabbitClient();
```

Comment:
This can leave rabbitClient and rabbitManagementTemplate as null. I expect that's ok since you could have the case where this health check runs before RabbitMQ comes up. However, I would think if that's the case, we could fail fast here and report everything as unhealthy. Thoughts?

Reply:
Doing that could also remove a bunch of null checks in the code called from the rest of this method.

Reply:
OK, this is fair.  I updated runHealthCheck to skip the health checks if the client & template are null.  I also updated the isHealthy method to ensure that rabbitClient, and rabbitManagementTemplate are non-null.  This should account for any failure to connect to rabbit at initialization.  The redundant null checks should have been removed as well.  

I also updated the recover method to perform a similar check on the client and management template.  So, that should be cleaner now as well.